Improvements

meengit opened the pull request:
Hi @kaste, I removed all default selectors (https://github.com/meengit/SublimeLinter-contrib-codeclimate/pull/5/commits/074eb5c2a151d83b8012ed39734825a5cb432278) and updated the documentation (https://github.com/meengit/SublimeLinter-contrib-codeclimate/pull/5/commits/6271cd174da2fab6c1840494a64d00b09ef32518) and the installation message (https://github.com/meengit/SublimeLinter-contrib-codeclimate/pull/5/commits/dd7312f7619f60cca9e63c0c0f3cb516c3fe32d4). As far as I can see, the changes are taking place as expected, and the linter is still working if I add, for example `source.php` to my global linter settings.

Do you think this is a practical solution?

meengit committed:
ADD: Restricted build branches

meengit committed:
MOD: Replaced .sublimelinterrc with flake8 conf

meengit committed:
DEL: Emptied default selector

meengit committed:
MOD: Simplified the documentation

meengit committed:
MOD: Updated license

meengit committed:
MOD: Updated message

meengit committed:
MOD: Fixed python 3.3

meengit committed:
MOD: Directly run on python 3.8

kaste commented:
Erm, `'selector': ()` (t.i. setting to an empty tuple) should throw at runtime *unless* a user overrides it in their settings.  Doesn't it?

I suggest you prepare and merge the trivial fixes for example changing the links and fixing CI/travis and then prepare the transition and release.  

I actually had no ideas, just wondering how codeclimate supports so many languages.  What happens if you try to lint a file for which no backend code has been downloaded yet?  Can you check if a file/language is supported?  Is the config file mandatory and could serve as a marker to enable this plugin?  Any other "markers" to automatically enable/disable this plugin? 

If it is really too slow, you should maybe suggest a user sets `disable: true` in the global settings, and `disable: false` per project.  If you instead want to narrow the language triggers down, we maybe need a hack/idea because *not* setting any selector is not supported.  (Usually we have a wide selector, and then for example `raise PermanentError` at some point.)

meengit committed:
MOD: Reverted default selectors

meengit committed:
ADD: Performance warning & limitations

meengit commented:
Hi @kaste

Oh… yes, I see the error in SublimeText's console now.

For a short time, I thought about something like:

```python
defaults = {
        'selector': (
            'source.xoo '
        )
    }
```

But then I thought this is kind of unclear, so I switched (back) to the default selectors and updated the installation message and the README.md with a performance warning. Do you think the current solution is understandable?

If you have many files and projects open simultaneously in SublimeText, performance can become an issue because codeclimate always starts some docker containers to run its tasks, which can become resource-intensive.

I've already updated the links and Travis-ci (https://travis-ci.com/github/meengit/SublimeLinter-contrib-codeclimate) in this pull request. Because we have no merged pull request on `main`, the build status is still unknown. I also migrated to Python 3.8 in Travis-ci. Since we do only lintings at the moment, I see no problems with this. There are some issues with Python 3.3 in Travis-ci.

As far as I know and experienced, `codeclimate` tries to install missing engines automatically. However, it is even better to install them manually with `engines:install` before starting the analyzes. You can also list the already installed engines with `engines:list`. The config file isn't mandatory. Without, `codeclimate` will run it's minimal default analyzes (https://github.com/meengit/SublimeLinter-contrib-codeclimate/tree/improvements#how-it-works). I see some possibilities to detect if the linter should run or not. For instance, a `.codeclimate.yml` in the folder root could be such a marker. I suggest bringing the plugin "online" for now and adding this kind of handling as an enhancement in the next months.

kaste commented:
👍 LGTM!  🚢 it.

meengit committed:
MOD: Updates installation message
